src/components/JsonSchemaForm/index.tsx

```diff
@@ -0,0 +1,231 @@
+              id={key}
+              key={label}
+              fieldSize="md"
+              label={label}
+              value={schemaValues[label] || ""}
+              error={""}
+              onChange={(e) => {
+                handleChange(label, e.target.value);
+              }}
+            />
+          );
+        case "U32":
+        case "U64":
+        case "U128":
+        case "U256":
+          return (
+            <PositiveIntPicker
+              id={key}
+              key={label}
+              label={label}
+              value={schemaValues[label] || ""}
+              error={""}
+              onChange={(e) => {
+                handleChange(label, e.target.value);
+              }}
+            />
+          );
+
+        case "I32":
+        case "I64":
+        case "I128":
+        case "I256":
+          return null;
+        // prop.type
+        case "array":
+          // if the array has items, render the items
+          if (Array.isArray(prop.items)) {
+            // @todo add minLength and maxLength
+            return prop.items.map((item: any, index: number) =>
+              renderComponent(key, item, index),
+            );
+          }
+          // render a button to add an item to the array
+          return (
+            <Box gap="sm" key={label}>
+              <LabelHeading size="md">{label}</LabelHeading>
+              <div>
+                <Button
+                  variant="secondary"
+                  size="md"
+                  onClick={() => {}}
+                  type="button"
+                >
+                  Add {label}
+                </Button>
+              </div>
+            </Box>
+          );
+
+        default:
+          return null;
+      }
+    }
+  };
+
+  const render = (item: any): React.ReactElement => {
+    const argLabels = item.properties ? Object.keys(item.properties) : [];
+    const fields = item.properties ? Object.entries(item.properties) : [];
+
+    return (
+      <>
+        {renderTitle(name, argLabels, fields)}
+
+        <Box gap="md" key={name}>
+          {fields.map(([key, prop]) => (
+            <>{renderComponent(key, prop)}</>
+          ))}
+        </Box>
+      </>
+    );
+  };
+
+  return (
+    <Box gap="md">
+      <Card>{render(dereferencedSchema)}</Card>
+    </Box>
+  );
+};
+
+const getTupleLabel = (items: any[]): string => {
+  const typeLabels = items.map((item) => {
+    if (item.$ref) {
+      // Handle single reference type (e.g., Address)
+      return item.$ref.replace("#/definitions/", "");
+    }
+    if (item.type === "array" && item.items?.$ref) {
+      // Handle array type (e.g., vec<Address>)
+      const innerType = item.items.$ref.replace("#/definitions/", "");
+      return `vec<${innerType}>`;
+    }
+    return item.type;
+  });
+
+  return `tuple<${typeLabels.join(",")}>`;
+};
+
+const renderTitle = (
+  name: string,
+  labels: string[],
+  fields: [string, any][],
+) => {
+  if (labels) {
```

Comment:
To make sure the array is not empty.

```suggestion
  if (labels.length) {
```